### DISCUSSION TITLE
export 키워드를 사용하는 예시를 작성해주세요

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#103] [모듈]
- export 키워드를 사용하는 예시를 작성해주세요 
- 그와 해당되는 import 예시도 적어주세요
- export, default export 의 차이점은 무엇인가요?

**Comment by publdaze** [comment]
> * export 키워드를 사용하는 예시를 작성해주세요
```js
// something.js
export const a = 1;
export function b() {}
```
```js
// something.js
const a = 1;
function b() {}

export { a, b };
```
> * 그와 해당되는 import 예시도 적어주세요
```js
import { a, b } from 'something.js'
```
```js
import { a as myA, b as myB } from 'something.js'
```

> * export, default export 의 차이점은 무엇인가요?
- named export는 여러 식별자를 묶어 한 번에 export 가능, import 할 때 중괄호로 묶어 각각의 식별자를 정확히 명시해야 import 가능
- default export는 한 파일에 하나만 가능, default export 뒤에 var, let, const 키워드가 올 수 없고 바로 식별자 혹은 값 자체로 내보내야 함 (ex. `default export a = 1;`), import할 때 중괄호 사용하지 않고 별칭으로 import 가능

**Comment by pipisebastian** [comment]
> * export 키워드를 사용하는 예시를 작성해주세요
> * 그와 해당되는 import 예시도 적어주세요
- 1
```js
// lib.mjs
export const a = 1;
export const b = 2;
```
```js
import {a,b} from "./lib.mjs"
```
- 2
```js
// lib.mjs
const a = 1;
const b = 2;

export { a, b };
```
```js
import {a as myA, b as myB} from "./lib.mjs"
```
> export, default export 의 차이점은 무엇인가요?

- **`export` (named export)**
    - 여러 개의 식별자 export 가능합니다.
    - import 에서는, export 할 때와 동일한 이름을 사용해야하며, {} 중괄호로 묶어 가져와야합니다.
    - `import { name } from './module.js';`
- **`default export`**
    - 하나의 값만 export 가능합니다. 이때, 식별자를 export 해도 되지만, 값 자체만을 export해도 됩니다. ex) `export default true;`
    - import 에서는, {} 중괄호 없이, 임의이 이름으로 import가 가능합니다.
    - `import defaultName from './module.js';`
